Change

MA33AI opened the pull request:
Add new packages "Controller" and "Enumiration"change method "getEmployeeById"

MA33AI committed:
Add new packages "Controller" and "Enumiration"change method "getEmployeeById"

TheLostRu commented on src/main/java/com/example/demo/spring/components/controller/PracticeRestController.java:
```diff
@@ -1,15 +1,14 @@
 import org.springframework.web.bind.annotation.*;
 
-import javax.swing.text.html.parser.Entity;
 import java.util.*;
```
убери в настройках идеи ваилдкарт импорты и нажми ctrl + L

TheLostRu commented on src/main/java/com/example/demo/spring/components/controller/PracticeRestController.java:
```diff
@@ -1,15 +1,14 @@
-import javax.swing.text.html.parser.Entity;
 import java.util.*;
 
 //@CrossOrigin(origins = "http://localhost:8081")
```
старайся не держать комментированный код если он не нужен прям

TheLostRu replied:
по инжекшну не могу строчку отметить тк она не в чейнджесах, но убери мапу ненужную и инжект сделай через конструктор, а то зачем у тебя ламбок аннотация тогда

TheLostRu commented on src/main/java/com/example/demo/spring/components/controller/PracticeRestController.java:
```diff
@@ -30,7 +29,7 @@ public ResponseEntity<List<Employee>> getAllEmployees(@RequestParam("name") Stri
             List<Employee> employees = new ArrayList<Employee>();
```
во 1 нужно сменить ResponseEntity<List<Employee>> просто на лист, в респонс спринг сам обернёт, во 2 раздели логику получения 1 и всех работяг на 2 разных метода, в 3 вместо траев всяких и тд сделай лучше отброс ошибки и напиши обработчик ошибок спринговый (@RestControllerAdvice)

TheLostRu commented on src/main/java/com/example/demo/spring/entitys/Employee.java:
```diff
@@ -1,6 +1,6 @@
 package com.example.demo.spring.entitys;
 
-import com.example.demo.spring.components.StatusEmployee;
```
пакет переименуй в controller и старайся пакеты называть в единственном числе

MA33AI committed:
1) Delete all comments
2) Delete map for save objects not in DataBase
3) Delete annotation @Autowired over field (employeeRepository)
4) Change in code ResponseEntity<List> to List<>
5) Change try/catch construction to (throw new NotFoundException)
6) Change names packages to the  singular
7) Add Service layer and move part code from EmployeeRestController to EmployeeService.
8) Delete annotations @Column and @Table in Employee and add @Enumerated over field (status)
9) Add method (findByName) in EmployeeRepository
10) Change type Integer to UUID on field (id) in Employee
